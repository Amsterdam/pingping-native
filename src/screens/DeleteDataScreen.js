import React, {useState} from 'react';

import {useMutation} from '@apollo/client';
import AsyncStorage from '@react-native-community/async-storage';
import PropTypes from 'prop-types';
import {StyleSheet, Text, View, ScrollView} from 'react-native';

import {testIDs} from '../../e2e/modulesTestIDs';
import {resetStore} from '../apollo/apolloClient';
import DELETE_USER_MUTATION from '../apollo/Mutation/deleteUserMutation';
import TrashIcon from '../assets/svg/icons/TrashIcon';
import FilledHeader from '../components/header/FilledHeader';
import ContentLayout from '../components/layout/ContentLayout';
import DeleteDataModal from '../components/modals/DeleteDataModal';
import Container from '../components/shared/Container';
import Button from '../components/shared/RoundedButton';
import Body from '../components/typography/Body';
import Title from '../components/typography/Title';
import theme from '../config/theme';
import sentryHelper from '../helpers/sentryHelper';

const DeleteDataScreen = ({navigation, setLogOut}) => {
	const [open, setOpen] = useState(false);
	const [loading, setLoading] = useState(false);
	const [deleteUser] = useMutation(DELETE_USER_MUTATION);

	const doDeleteUser = async () => {
		setLoading(true);
		try {
			await deleteUser({
				variables: {
					confirm: 'delete',
				},
			});
			await AsyncStorage.clear();
			setLogOut();
			resetStore();
		} catch (error) {
			setLoading(false);
			sentryHelper(error.message);
		}
	};

	return (
		<Container
			testID={testIDs.DELETE_DATA.SCREEN}
			statusBarColor={theme.colors.headerColor}>
			<FilledHeader navigation={navigation} title="Privacy" />
			<ScrollView contentContainerStyle={styles.scrollView}>
				<ContentLayout>
					<Title style={styles.margin}>Jouw eigen gegevens</Title>
					<View style={styles.emojiContainer}>
						<Text>ðŸ‘†ðŸ˜Œ</Text>
					</View>
					<Body variant="b3" style={styles.margin}>
						Wij van Ping Ping gebruiken de door jouw ingevulde gegevens alleen
						om jouw route te bepalen en de app te verbeteren. Wij zullen nooit
						jouw gegevens verkopen aan andere partijen.
					</Body>
					<Body variant="b3" style={styles.margin}>
						Wil je toch graag je inloggegevens verwijderen? Druk simpelweg op de
						knop hieronder om je gegevens uit ons systeem te halen. Hierdoor
						gaat je voortgang verloren.
					</Body>
					<View style={styles.inputContainer}>
						<Button
							deleteButton
							icon={<TrashIcon height={20} color={theme.colors.white} />}
							label="Verwijder mijn gegevens"
							onPress={() => setOpen(true)}
							testid={testIDs.DELETE_DATA.DELETE_BUTTON}
							full
						/>
					</View>
				</ContentLayout>
			</ScrollView>
			<DeleteDataModal
				open={open}
				setOpen={setOpen}
				setLogOut={setLogOut}
				navigation={navigation}
				doDeleteUser={doDeleteUser}
				loading={loading}
			/>
		</Container>
	);
};

const styles = StyleSheet.create({
	margin: {
		marginBottom: 30,
	},
	scrollView: {
		backgroundColor: theme.colors.white,
		flex: 1,
	},
	inputContainer: {alignItems: 'center', justifyContent: 'center'},
	emojiContainer: {
		alignItems: 'center',
		marginBottom: 10,
	},
});

DeleteDataScreen.propTypes = {
	navigation: PropTypes.object.isRequired,
	setLogOut: PropTypes.func.isRequired,
};

export default DeleteDataScreen;
