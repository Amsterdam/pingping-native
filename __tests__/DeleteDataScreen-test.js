import React from 'react';

import {MockedProvider} from '@apollo/client/testing';
import renderer from 'react-test-renderer';

import DeleteDataScreen from '../src/screens/DeleteDataScreen';

jest.useFakeTimers();
const navigation = {navigate: jest.fn()};
const setLogOut = jest.fn();

test('renders correctly', () => {
	const tree = renderer
		.create(
			<MockedProvider mocks={[]}>
				<DeleteDataScreen navigation={navigation} setLogOut={setLogOut} />
			</MockedProvider>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
