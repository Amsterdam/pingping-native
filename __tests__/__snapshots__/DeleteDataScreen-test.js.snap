// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
  testID="DELETE_DATA_SCREEN"
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#0d2036",
        }
      }
    >
      <View
        noShadow={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 0.5,
              "elevation": 0,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 6,
              "paddingRight": 10,
              "paddingTop": undefined,
              "right": 0,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
              "top": 0,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#0d2036",
            },
          ]
        }
        transparent={true}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            forwardedRef={[Function]}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 4,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID="HEADER_BACK_BUTTON"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 34,
                      "marginLeft": 1,
                      "marginRight": 2,
                      "marginTop": 0,
                      "paddingTop": 1,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ],
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="AntDesign"
            >
              î˜œ
            </Text>
          </View>
        </View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "Heavitas",
              },
              Object {
                "fontSize": 26,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "#fff",
                "marginTop": 5,
              },
            ]
          }
          uppercase={false}
        >
          Privacy
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
              "padding": 25,
              "position": "relative",
              "width": 750,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "fontFamily": "Heavitas",
                },
                Object {
                  "fontSize": 58,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "marginBottom": 30,
                },
              ]
            }
            uppercase={false}
          >
            Jouw eigen gegevens
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginBottom": 10,
              }
            }
          >
            <Text>
              ðŸ‘†ðŸ˜Œ
            </Text>
          </View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "Raleway-Regular",
                  "lineHeight": 46,
                },
                Object {
                  "fontSize": 29,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "marginBottom": 30,
                },
              ]
            }
          >
            Wij van Ping Ping gebruiken de door jouw ingevulde gegevens alleen om jouw route te bepalen en de app te verbeteren. Wij zullen nooit jouw gegevens verkopen aan andere partijen.
          </Text>
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "Raleway-Regular",
                  "lineHeight": 46,
                },
                Object {
                  "fontSize": 29,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "marginBottom": 30,
                },
              ]
            }
          >
            Wil je toch graag je inloggegevens verwijderen? Druk simpelweg op de knop hieronder om je gegevens uit ons systeem te halen. Hierdoor gaat je voortgang verloren.
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              forwardedRef={[Function]}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#f9454f",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 10,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
              testID="DELETE_DATA_BUTTON"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#fff",
                          "fontSize": 24,
                          "marginLeft": 16,
                          "marginRight": 16,
                          "paddingTop": 2,
                        },
                        Object {
                          "color": "#fff",
                          "fontSize": 44,
                        },
                      ],
                      Object {
                        "fontFamily": "anticon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  type="AntDesign"
                >
                  îšŸ
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "System",
                        "fontSize": 16.5,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      Object {
                        "color": "#fff",
                        "fontFamily": "Heavitas",
                        "fontSize": 26,
                        "paddingTop": 3,
                      },
                      Object {},
                    ]
                  }
                  uppercase={false}
                >
                  Verwijder mijn gegevens
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(1,1,1,0.8)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="DELETE_DATA_MODAL"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "elevation": 5,
            "justifyContent": "center",
            "margin": 20,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <View
          style={
            Object {
              "padding": 35,
              "position": "relative",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            forwardedRef={[Function]}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 57,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "position": "absolute",
                "right": 0,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 24,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "paddingTop": 2,
                    },
                    Object {
                      "color": "#000",
                      "fontSize": 44,
                    },
                  ],
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="AntDesign"
            >
              î˜³
            </Text>
          </View>
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "fontFamily": "Heavitas",
                },
                Object {
                  "fontSize": 58,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "marginBottom": 25,
                },
              ]
            }
            uppercase={false}
          >
            Gegevens Verwijderen
          </Text>
          <Text
            ellipsizeMode="tail"
            numberOfLines={0}
            style={
              Array [
                Object {
                  "fontFamily": "Raleway-Regular",
                  "lineHeight": 46,
                },
                Object {
                  "fontSize": 29,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "marginBottom": 20,
                },
              ]
            }
          >
            Weet je zeker dat je jouw gegevens wilt verwijderen? Wanneer je de gegevens hebt verwijdered is er geen mogelijkheid meer om deze terug te halen..
          </Text>
          <View>
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              forwardedRef={[Function]}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#fb9f00",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 10,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "marginVertical": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
              testID=""
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "System",
                        "fontSize": 16.5,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      Object {
                        "color": "#fff",
                        "fontFamily": "Heavitas",
                        "fontSize": 26,
                        "paddingTop": 3,
                      },
                      Object {},
                    ]
                  }
                  uppercase={false}
                >
                  Niet verwijderen
                </Text>
              </View>
            </View>
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              forwardedRef={[Function]}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#e9e9e9",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 10,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "marginVertical": 5,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
              testID="DELETE_DATA_MODAL_BUTTON"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#000",
                          "fontSize": 30,
                        },
                        Object {
                          "color": "#fff",
                          "fontSize": 24,
                          "marginLeft": 16,
                          "marginRight": 16,
                          "paddingTop": 2,
                        },
                        Object {
                          "color": "#fff",
                          "fontSize": 44,
                        },
                      ],
                      Object {
                        "fontFamily": "anticon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  type="AntDesign"
                >
                  îšŸ
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "System",
                        "fontSize": 16.5,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      Object {
                        "color": "#fff",
                        "fontFamily": "Heavitas",
                        "fontSize": 26,
                        "paddingTop": 3,
                      },
                      Object {},
                    ]
                  }
                  uppercase={false}
                >
                  Verwijder mijn gegevens
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
